optuna_dashboard/ts/components/GraphIntermediateValues.tsx

```diff
@@ -51,20 +51,31 @@ const plotIntermediateValue = (trials: Trial[], mode: string) => {
     return
   }
 
-  const filteredTrials = trials.filter(
-    (t) =>
-      t.state === "Complete" ||
-      (t.state === "Pruned" && t.values && t.values.length > 0)
-  )
-  const plotData: Partial<plotly.PlotData>[] = filteredTrials.map((trial) => {
-    const values = trial.intermediate_values.filter((iv) => iv.value !== "inf")
-    return {
-      x: values.map((iv) => iv.step),
-      y: values.map((iv) => iv.value),
-      mode: "lines+markers",
-      type: "scatter",
-      name: `trial #${trial.number}`,
+  const sortedFilteredTrials = trials
+    .filter(
+      (t) =>
+        t.state === "Complete" ||
+        (t.state === "Pruned" && t.values && t.values.length > 0) ||
+        t.state == "Running"
+    )
+    .slice()
+    .sort((a: Trial, b: Trial) => a.number - b.number)
```

Comment:
I have a question about sorting. In my recognition, API server returns trials in ascending order and `.filter()` method in JavaScript keeps the order. Do we need to sort trials here?

Reply:
Sorry, the following comment was still pending. I have just removed it.